math/mathcore/inc/Math/Util.h

```diff
@@ -66,35 +65,64 @@ namespace ROOT {
 
    } // end namespace Util
 
+   ///\class KahanSum
+   /// The Kahan compensate summation algorithm significantly reduces the numerical error in the total obtained
+   /// by adding a sequence of finite precision floating point numbers.
+   /// This is done by keeping a separate running compensation (a variable to accumulate small errors).\n
+   ///
+   /// The intial values of the result and the correction are set to the default value of the type it hass been
+   /// instantiated with.\n
+   /// ####Examples:
+   /// ~~~{.cpp}
+   /// std::vector<double> numbers = {0.01, 0.001, 0.0001, 0.000001, 0.00000000001};
+   /// ROOT::Math::KahanSum<double> k;
+   /// k.Add(numbers);
+   /// ~~~
+   /// ~~~{.cpp}
+   /// auto result = ROOT::Math::KahanSum<double>::Accumulate(numbers);
+   /// ~~~
    template <class T>
    class KahanSum {
    public:
+      /// Constructor accepting a initial value for the summation as parameter
+      KahanSum(const T &initialValue = T{}) : fSum(initialValue) {}
+
+      /// Single element accumulated addition.
       void Add(const T &x)
       {
-         auto y = x - correction;
-         auto t = sum + y;
-         correction = (t - sum) - y;
-         sum = t;
+         auto y = x - fCorrection;
+         auto t = fSum + y;
+         fCorrection = (t - fSum) - y;
+         fSum = t;
       }
 
-      void Add(const std::vector<T> &elements)
+      /// Iterate over a datastructure referenced by a pointer and accumulate on the exising result
+      template <class Iterator>
+      void Add(const Iterator begin, const Iterator end)
```

Comment:
@xvallspl, @lmoneta  : is it really better to have begin and end here? Can't we pass something to iterate on? With range based looping this would not imply any code change in the body of the methods.